### PR TITLE
docs: Improvement regarding accel on windows

### PR DESCRIPTION
**Pull request by ciarancourtney**
Some docs improvments related to accel arg, particularly on windows hosts

### CODE REVIEW

**Comment by vheon** [comment]
why is setting the `qemuargs` directly is better then setting:

```hcl
machine_type = "q35"
accelerator = "hvf:kvm:whpx:tcg"
```

??

**Comment by ciarancourtney** [comment]
> why is setting the `qemuargs` directly is better then setting:
> 
> ```terraform
> machine_type = "q35"
> accelerator = "hvf:kvm:whpx:tcg"
> ```
> 
> ??

Ultimately that would be better but its not supported in the current version, you can only set one `accelerator`. I had thought about submitting a PR to fallback to all possible accelerators and let qemu take the lead but thought it would introduce too much magic.

**Comment by vheon** [comment]
@ciarancourtney I didn't tried and only now I see this code here

https://github.com/hashicorp/packer-plugin-qemu/blob/8ad4058e86f6a4b36bd49434ae7deed7c6ecd5c8/builder/qemu/config.go#L810-L813

maybe if we do a simple `split` on the passed string and check that each section should be one of the supported one could work. Something like:

```golang
for _, accelerator := range strings.Split(c.Accelerator, ":") {
		if _, ok := accels[accelerator]; !ok {
			errs = packersdk.MultiErrorAppend(
			errs, errors.New("invalid accelerator, only 'kvm', 'tcg', 'xen', 'hax', 'hvf', 'whpx', or 'none' are allowed"))
		}
	}
```

probably `none` should be handled differently...

File: builder/qemu/config.go [hunk 1]
```diff
@@ -235,9 +235,13 @@ type Config struct {
 	// add [ "-global", "virtio-pci.disable-modern=on" ] to `qemuargs` depending on the
 	// guest operating system.
 	//
-	// ~> For `whpx`, note that [Stefan Weil's QEMU for Windows distribution](https://qemu.weilnetz.de/w64/)
```
Comment (nywilken) [inline builder/qemu/config.go:238]: Why is this removed?

**Comment by nywilken** [comment]
This change also requires the execution of `go generate ./...` to update the generated docs.

**Comment by lbajolet-hashicorp** [comment]
Regarding the acceleration methods, there's two alternatives:

* Either we do as @vheon proposed, and allow the `:` syntax and validate that each are supported.
* We add a new attribute (`accelerators` for example), which will let us define multiple in order of preference, and pass it to qemu as a string with `:` in between each provided accelerator.

I agree with @nywilken though, this should probably be tracked in a separate issue.

Regarding `none`, as it stands, it's essentially to explicitely not define the `accelerator` attribute, which amounts to qemu's defualt behaviour, i.e. `tcg`.
When specifying a series of accelerators then, `none` should not be a valid choice, and should be rejected (or at worst ignored with a warning specifying why).

File: builder/qemu/config.go (hunk 1 above)
Comment (lbajolet-hashicorp) [reply builder/qemu/config.go:238]: According to the commit that changes this, WHPX is supported by Stefan Weil's Windows qemu distribution, so in that sense I tend to agree that this can be removed from our docs.
I cannot test this however since I don't have a Windows machine on hand

**Review by lbajolet-hashicorp — APPROVED** [review]
Hi @ciarancourtney,

Thanks for opening this, and sorry it was left open for so long.

I've taken a new look at the code, rebased it, and re-generated the docs from the changes in the code.
For future reference, if you want to suggest more changes to the documentation later, plase run `make generate` so the partials are up-to-date, otherwise we have some actions that will fail.

I'll merge this once tests go green, and we can keep an issue open for the accelerator choices. As pointed out in your commit, the `:` notation is now deprecated (refer to [this thread](https://mail.gnu.org/archive/html/qemu-discuss/2019-11/msg00115.html) for more information), so we should probably consider using multiple `-accel` options if/when we want to support this feature.

This being said, thanks again for opening this one @ciarancourtney!
